lutris/util/display.py

```diff
@@ -130,37 +130,121 @@ def get_display_manager():
 USE_DRI_PRIME = len(_get_graphics_adapters()) > 1
 
 
-def get_compositor_commands():
-    """Nominated for the worst function in lutris"""
+# Pretend this is an enum.
+class DE:
+
+    """Enum of desktop environments."""
+
+    PLASMA = object()
+    MATE = object()
+    XFCE = object()
+    DEEPIN = object()
+    UNKNOWN = object()
+
+
+def get_desktop_environment():
```

Comment:
I don't recall where but I think we already have a function that determines what the current desktop environment is. Maybe @strycore knows where it is.

Reply:
`grep -iF DESKTOP_SESSION -r .` did not yield any other place where DESKTOP_SESSION is read, so I don't think there's any other place.

Reply:
False alarm, it was this function that you are rewriting.